tst/PeerConnectionFunctionalityTest.cpp

```diff
@@ -599,6 +599,115 @@ TEST_F(PeerConnectionFunctionalityTest, connectTwoPeersExpectFailureBecauseNoCan
     freePeerConnection(&answerPc);
 }
 
+TEST_F(PeerConnectionFunctionalityTest, noLostFramesAfterConnected)
+{
+    struct Context {
+        MUTEX mutex;
+        ATOMIC_BOOL done;
+        CVAR cvar;
+    };
+
+    RtcConfiguration configuration;
+    Context context;
+    PRtcPeerConnection offerPc = NULL, answerPc = NULL;
+    RtcMediaStreamTrack offerVideoTrack, answerVideoTrack, offerAudioTrack, answerAudioTrack;
+    PRtcRtpTransceiver offerVideoTransceiver, answerVideoTransceiver, offerAudioTransceiver, answerAudioTransceiver;
+    RtcSessionDescriptionInit sdp;
+    ATOMIC_BOOL seenFirstFrame = FALSE;
+    Frame videoFrame;
+
+    MEMSET(&configuration, 0x00, SIZEOF(RtcConfiguration));
+    MEMSET(&videoFrame, 0x00, SIZEOF(Frame));
+
+    videoFrame.frameData = (PBYTE) MEMALLOC(1);
+    videoFrame.size = 1;
+    videoFrame.presentationTs = 0;
+
+    context.mutex = MUTEX_CREATE(FALSE);
+    ASSERT_NE(context.mutex, INVALID_MUTEX_VALUE);
+    context.cvar = CVAR_CREATE();
+    ASSERT_NE(context.cvar, INVALID_CVAR_VALUE);
+    ATOMIC_STORE_BOOL(&context.done, FALSE);
+
+    EXPECT_EQ(createPeerConnection(&configuration, &offerPc), STATUS_SUCCESS);
+    EXPECT_EQ(createPeerConnection(&configuration, &answerPc), STATUS_SUCCESS);
+
+    addTrackToPeerConnection(offerPc, &offerVideoTrack, &offerVideoTransceiver, RTC_CODEC_VP8, MEDIA_STREAM_TRACK_KIND_VIDEO);
+    addTrackToPeerConnection(answerPc, &answerVideoTrack, &answerVideoTransceiver, RTC_CODEC_VP8, MEDIA_STREAM_TRACK_KIND_VIDEO);
+
+    auto onFrameHandler = [](UINT64 customData, PFrame pFrame) -> void {
+        UNUSED_PARAM(pFrame);
+        if (pFrame->frameData[0] == 1) {
+            ATOMIC_STORE_BOOL((PSIZE_T) customData, 1);
+        }
+    };
+    EXPECT_EQ(transceiverOnFrame(answerVideoTransceiver, (UINT64) &seenFirstFrame, onFrameHandler), STATUS_SUCCESS);
+
+    auto onICECandidateHdlr = [](UINT64 customData, PCHAR candidateStr) -> void {
```

Comment:
I can't use `connectTwoPeers` because the test is trying to exercise the `CONNECTED` state without knowing the other peer's state.